browser/components/MarkdownPreview.js

```diff
@@ -1023,7 +1023,8 @@ export default class MarkdownPreview extends React.Component {
 
     if (!href) return
 
-    const regexNoteInternalLink = /main.html#(.+)/
+    const regexNoteInternalLink = process.env.NODE_ENV === 'production' ? /main.production.html#(.+)/ : /main.development.html#(.+)/
```

Comment:
This is the regex I've metioned.
After having a second look on this I think it's important to update. I tested the regex [here](https://regex101.com/r/zRLTIR/1).

Reply:
Hmm.. `/main.[A-Za-z0-9]+.html#/` might be better to have rather than two regex.

Reply:
Hmm.. /main.[A-Za-z0-9]+.html#/ might be better to have rather than two regex.

Reply:
Yes, good idea. Maybe also `/main[.\w]*.html#/` will work - also supports `main.html`. I'll update this later today.

Reply:
https://regex101.com/r/zRLTIR/2

Reply:
@AWolf81 Yeah I think your idea is better. Could you fix it too? Then I think we could merge this.

Reply:
@Rokt33r I don't get it to work. Seems like my previous regex was also not working.
Here is my code inside `MarkdownPreview.js`:
![grafik](https://user-images.githubusercontent.com/3046542/57505548-0ce0e800-72f9-11e9-967f-876e075c2a33.png)

And I'm getting the following output on local link click:
![grafik](https://user-images.githubusercontent.com/3046542/57505581-313cc480-72f9-11e9-864f-5b5f190d1368.png)

Seems like I'm not getting into the if for local file links.

Reply:
I think it's because there is no `main.development.html` in the `linkHash` string. Maybe new router changed here.
I have to check how to detect the local link.

Reply:
Maybe I have the link in markdown not right. Do I need one slash in the url? I tested with `[test link](#Test)`. I was a bit in a hurry before going to work. I'll check it later today after work.

Reply:
I'll try it too later.

Reply:
OK, I've fixed it by modifying parsing of the href string. Inspired by the following [post](https://www.abeautifulsite.net/parsing-urls-in-javascript).

Tested all link types (empty link, line link `:line:<number>`, tag link `#title` & note link `:note:noteKey`). I just couldn't verify old link style `storage.key-note.key` - not sure how to test it. But it should still work.

**Note:** Adding a slash to a tag link e.g. `/#title` won't work but I think that's OK - same behavior as on master.

Reply:
I think `ctrl+left mouse` handling in the editor is not working for every link type, right? Tested on master. I think it's using just the open external Url link - it would be good to share the link handling code for this location so the same functionality is available there.

Should we create an issue for this so it can be improved later?